spec/controllers/jobs_controller_spec.rb

```diff
@@ -30,6 +33,24 @@
     it "renders the :index template" do
       expect(response).to render_template :index
     end
+
+    context "with a search query" do
+      before :each do
+        get :index, q: "foobar"
+      end
+
+      it "includes job when the query matches its title" do
+        expect(assigns(:jobs)).to include matching_title_job
```

Comment:
Moving this logic to the model makes testing more straightforward too. Tests here should probably exercise passing the optional search param only.

When setting up test data it's good practice to make them related to the problem domain. "Foo" and "bar" are okay if, for example, you're exercising a feature on plain strings. Here "backend developer" or "Québec" would make more sense as jobs titles.